README.md

```diff
@@ -168,3 +168,73 @@ INFO: Build completed successfully, 1825 total actions
 
 INFO: Build completed successfully, 1825 total actions
 ```
+
+# Using external packages
+
+External packages can be specified using `dbx_py_pypi_piplib`, which can be imported from `@dbx_build_tools//build_tools/py:py.bzl`. Consider a python file which uses `import some_library`, and you wish for this to be retrieved through PyPI. When `bzl` reads this import, it will try to find its definition at `//pip/some_library`; if it finds it, it will be added to the `deps` list of the `dbx_py_binary` or `dbx_py_library` that depends on it in the corresponding `BUILD` file.
+
+## If the external package has no dependencies
+If `some_library` has no external dependencies, you can thus create a file at `[project_root]/pip/some_library/BUILD` containing
+```
+load("@dbx_build_tools//build_tools/py:py.bzl", "dbx_py_pypi_piplib")
+
+dbx_py_pypi_piplib(
+    name = "some_library",
+    pip_version = "1.2.3", #replace with the required version
+    visibility = ["//visibility:public"],
+)
+```
+and the build of your project will download and install the correct version, making it available to your code.
+
+## If the external package has runtime dependencies
+If, however, the external project has an external runtime dependency on a library called `another_library`, you should also create `[project_root]/pip/another_library/BUILD` along the same lines as the previous example. Then you can add it to a `deps` parameter to the `dbx_py_pypi_piplib` function for `some_library`:
+```
+# [project_root]/pip/another_library/BUILD:
+
+load("@dbx_build_tools//build_tools/py:py.bzl", "dbx_py_pypi_piplib")
+
+dbx_py_pypi_piplib(
+    name = "some_library",
```

Comment:
There's a typo here. Maybe instead, introduce `another_library` that depends on `some_library`. That will cut down on needing to duplicate.

Reply:
My apologies for the typo - I entirely agree with the suggestion of simplifying the instructions by using another_library as the dependent package rather than the dependency.